chainer/dataset/download.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import print_function
+import hashlib
+import os
+import shutil
+import tempfile
+
+import filelock
+from six.moves.urllib import request
+
+
+_dataset_root = os.environ.get('CHAINER_DATASET_ROOT',
+                               os.path.expanduser('~/.chainer/dataset'))
+
+
+def get_dataset_root():
+    """Gets the path to the root directory to download and cache datasets.
+
+    Returns:
+        str: The path to the dataset root directory.
+
+    """
+    return _dataset_root
+
+
+def set_dataset_root(path):
+    """Sets the root directory to download and cache datasets.
+
+    There are two ways to set the dataset root directory. One is by setting the
+    environmant variable ``CHAINER_DATASET_ROOT``. The other is by using this
+    function. The default dataset root is ``$HOME/.chainer/dataset``.
+
+    Args:
+        path (str): Path to the new dataset root directory.
+
+    """
+    global _dataset_root
+    _dataset_root = path
+
+
+def get_dataset_directory(dataset_name, create_directory=True):
+    """Gets the path to the directory of given dataset.
+
+    The generated path is just a concatenation of the global root directory
+    (see :func:`set_datasets_root` for how to change it) and the dataset name.
+    The dataset name can contain slashes, which are treated as path separators.
+
+    Args:
+        dataset_name (str): Name of the dataset.
+        create_directory (bool): If True (default), this function also creates
+            the directory at the first time. If the directory already exists,
+            then this option is ignored.
+
+    Returns:
+        str: Path to the dataset directory.
+
+    """
+    path = os.path.join(_dataset_root, dataset_name)
+    if create_directory:
+        try:
+            os.makedirs(path)
```

Comment:
How about setting `exist_ok` to `True` instead of catching OSError and grasping it.